tools/run_tests/pre_build_csharp.bat

```diff
@@ -38,8 +38,21 @@ cd /d %~dp0\..\..
 set NUGET=C:\nuget\nuget.exe
 
 if exist %NUGET% (
+  @rem Restore Grpc packages by packages since Nuget client 3.4.4 doesnt support restore
```

Comment:
But the way I'm restoring packages actually had to change to what's in https://github.com/grpc/grpc/pull/7568, in order to be work with both nuget 2.8 and nuget 3.4. Sorry for the out of date files here, but I was planning on using the pre-build scripts in that PR just to switch the nuget clients over on jenkins, so this could work